ksqldb-rocksdb-config-setter/src/test/java/io/confluent/ksql/rocksdb/KsqlBoundedMemoryRocksDBConfigTest.java

```diff
@@ -106,4 +128,111 @@ public void shouldDefaultIndexFilterBlockRatioConfig() {
+        is(CompactionStyle.UNIVERSAL.name()));
+  }
+
+  @Test
+  public void shouldDefaultCompressionTypeConfig() {
+    // Given:
+    final Map<String, Object> configs = ImmutableMap.of(
+        "ksql.plugins.rocksdb.cache.size", CACHE_SIZE
+    );
+
+    // When:
+    final KsqlBoundedMemoryRocksDBConfig pluginConfig = new KsqlBoundedMemoryRocksDBConfig(configs);
+
+    // Then:
+    assertThat(
+        pluginConfig.getString(KsqlBoundedMemoryRocksDBConfig.COMPRESSION_TYPE),
+        is(CompressionType.NO_COMPRESSION.name()));
+  }
+
+  @Test
+  public void shouldDefaultTrivialMove() {
+    // Given:
+    final Map<String, Object> configs = ImmutableMap.of(
+        "ksql.plugins.rocksdb.cache.size", CACHE_SIZE
+    );
+
+    // When:
+    final KsqlBoundedMemoryRocksDBConfig pluginConfig = new KsqlBoundedMemoryRocksDBConfig(configs);
+
+    // Then:
+    assertThat(
+        pluginConfig.getBoolean(KsqlBoundedMemoryRocksDBConfig.COMPACTION_TRIVIAL_MOVE),
+        is(false));
+  }
+
+  @Test
+  public void shouldDefaultMaxNumberOfConcurrentJobs() {
+    // Given:
+    final Map<String, Object> configs = ImmutableMap.of(
+        "ksql.plugins.rocksdb.cache.size", CACHE_SIZE
+    );
+
+    // When:
+    final KsqlBoundedMemoryRocksDBConfig pluginConfig = new KsqlBoundedMemoryRocksDBConfig(configs);
+
+    // Then:
+    assertThat(
+        pluginConfig.getInt(KsqlBoundedMemoryRocksDBConfig.MAX_BACKGROUND_JOBS),
+        is(-1));
+  }
+
+  @Test
+  public void shouldThrowWithIncorrectCompactionStyle() {
+    // Given:
+    final Map<String, Object> configs = ImmutableMap.of(
+        "ksql.plugins.rocksdb.cache.size", CACHE_SIZE,
+        "ksql.plugins.rocksdb.compaction.style", CompactionStyle.FIFO.name()
+    );
+
+    // When:
+    final Exception e = assertThrows(
+        ConfigException.class,
+        () -> new KsqlBoundedMemoryRocksDBConfig(configs)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Invalid value FIFO for configuration ksql.plugins.rocksdb.compaction.style: "
+            + "String must be one of: UNIVERSAL, LEVEL"));
+  }
+
+  @Test
+  public void shouldThrowWithIncorrectCompressionType() {
+    // Given:
+    final Map<String, Object> configs = ImmutableMap.of(
+        "ksql.plugins.rocksdb.cache.size", CACHE_SIZE,
+        "ksql.plugins.rocksdb.compression.type", "FOO"
+    );
+
+    // When:
+    final Exception e = assertThrows(
+        ConfigException.class,
+        () -> new KsqlBoundedMemoryRocksDBConfig(configs)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Invalid value FOO for configuration ksql.plugins.rocksdb.compression.type: "
+            + "String must be one of: "
+            + "NO_COMPRESSION, SNAPPY_COMPRESSION, ZLIB_COMPRESSION, BZLIB2_COMPRESSION, "
+            + "LZ4_COMPRESSION, LZ4HC_COMPRESSION, XPRESS_COMPRESSION, ZSTD_COMPRESSION, "
+            + "DISABLE_COMPRESSION_OPTION, null"));
```

Comment:
What does it mean to set the compression type to null? Is that actually allowed? `DISABLE_COMPRESSION_OPTION` could also be confusing so there might be a case to disallow it, but seems fine either way.

Reply:
> What does it mean to set the compression type to null? Is that actually allowed? `DISABLE_COMPRESSION_OPTION` could also be confusing so there might be a case to disallow it, but seems fine either way.

I just exposed whatever was allowed ([here](https://javadoc.io/doc/org.rocksdb/rocksdbjni/6.8.1/org/rocksdb/CompressionType.html)), but I agree with you.

Reply:
Did you ever try passing `null` to a real instance? That value is added by ksql's config validator ([link](https://github.com/confluentinc/ksql/blob/40ab84e5cf48426dc78096e231ca97b494b1607e/ksqldb-common/src/main/java/io/confluent/ksql/configdef/ConfigValidators.java#L80)) and not one of the original enum values, so it's possible that it throws and error and needs to be removed from the list of valid values.

In general, it's easy to expand the list of allowable values in the future, but harder to narrow the list after a version has been released with a more permissive list. We can err on the side of having more restrictive configs if there's uncertainty about what will work.

Reply:
> Did you ever try passing `null` to a real instance? That value is added by ksql's config validator ([link](https://github.com/confluentinc/ksql/blob/40ab84e5cf48426dc78096e231ca97b494b1607e/ksqldb-common/src/main/java/io/confluent/ksql/configdef/ConfigValidators.java#L80)) and not one of the original enum values, so it's possible that it throws and error and needs to be removed from the list of valid values.
> 
> In general, it's easy to expand the list of allowable values in the future, but harder to narrow the list after a version has been released with a more permissive list. We can err on the side of having more restrictive configs if there's uncertainty about what will work.

I removed both `DISABLE_COMPRESSION_OPTION` and ` null` from the options.